api/openapi.yaml

```diff
@@ -0,0 +1,94 @@
+components:
+  schemas:
+    v1.Account:
+      properties:
+        account_number:
+          type: string
+        id:
+          maximum: 1.8446744073709552e+19
+          minimum: 0
+          type: integer
+        org_id:
+          type: string
+      type: object
+    v1.Pubkey:
+      properties:
+        account_id:
+          maximum: 1.8446744073709552e+19
+          minimum: 0
+          type: integer
+        body:
+          type: string
+        id:
+          maximum: 1.8446744073709552e+19
+          minimum: 0
+          type: integer
+        name:
+          type: string
+      type: object
+    v1.ResponseError:
+      properties:
+        msg:
+          type: string
+        request_id:
+          type: string
+      type: object
+servers:
+- description: Local development
+  url: http://0.0.0.0:{port}/api/{applicationName}
+  variables:
+    applicationName:
+      default: provisioning
+    port:
+      default: "8000"
+openapi: 3.0.0
+info:
+  license:
+    name: GPL-3.0
+  title: provisioning-api
+  version: 0.0.1
+
+paths:
+  /pubkeys/{ID}:
+    get:
+      operationId: getPubkeyByID
+      parameters:
+        - name: ID
+          in: path
+          required: true
+          description: unique pubkey id
+          schema:
+            type: integer
```

Comment:
Now, the bit question to solve here (@adiabramovitch @ezr-ondrej): how to represent database IDs. From what I've learned today, people incorrectly believe that JSON format does not support bigger integer types that int32. That is simply not true, JSON (as the format https://www.json.org) does not put any limits on numbers, they are stored in decimals with optionally exponent formatting. JSON Schema (that is a totally different thing) put some constraints as well as but it supports actually (signed) int64: https://swagger.io/specification/

Let's first look how others do it. SourcesAPI does is simply using strings:

```json
  "components": {
    "parameters": {
      "ID": {
        "name": "id",
        "in": "path",
        "description": "ID of the resource",
        "required": true,
        "schema": {
          "example": "12345",
          "pattern": "^\\d+$",
          "type": "string"
        }
      },
```

Frankly, I don't like this solution. Image Builder simply uses integer without format, I am not sure if int64 is the default.

Current state: In the database we use `BIGINT` for all IDs which is a signed integer 64bits. In models we have (incorrectly) `uint64`, I introduced this bug because I started the DDL with `BIGSERIAL` which is actually unsigned, but I changed it to `BIGINT` later.

My proposal for OpenAPI: let's use `integer` type with `int64` format. This will nicely map to the model `int64` as well as `BIGINT` in the database. There is absolutely no reason to use string hacks, unless we hit a wall with this solution (OpenAPI looks quite complex).

Reply:
Fixing in my PR.

Reply:
It sounds very reasonable to me to propagate the type in our DB to the spec, it sets the expectation we can have ids that grow over int32 and if the limit is not in json spec, it should be up to the caller to make sure their implementation of JSON supports int64